apps/edxapp/templates/services/nginx/configs/cms.conf.j2

```diff
@@ -34,6 +34,17 @@ server {
     try_files $uri @proxy_to_cms_app;
   }
 
+{% if edxapp_nginx_admin_ip_withelist | length > 0 %}
+  location /admin {
+
+    if ($http_x_forwarded_for !~ ({{ edxapp_nginx_admin_ip_withelist | join("|") }})) {
```

Comment:
I think `http_x_forwarded_for` is only set behind a proxy.

Reply:
You mean only if there is a haproxy/nginx/whatever in front of openshift ?

Reply:
From what I understand, `HTTP_X_FORWARDED_FOR` can be a list of IPs when you are behind a number of proxies. If not behind a proxy, you should probably use `HTTP_X_REAL_IP`.

Reply:
But is it possible to directly expose an openshift or kubernetes cluster ? I think it's always behind a proxy

Reply:
yes but even behind a proxy, http_x_forwarded_for is a list that grows and anybody can send whatever they want into it. We can use http_x_real_ip and specify that the openshift should be placed behind a proxy that is configured to set this header with the value of the real `remote_ip`?

Reply:
We have to use `http_x_forwarded_for` but keep in my mind that it's a comma separated list and the first occurence in the list is the real ip : https://github.com/symfony/http-foundation/blob/897bf05a4dd936cfd028d7d04f0f21590c024030/Request.php#L791-L798

Reply:
ok but if there is no proxy, http_x_forwarded_for is empty so you must default to `remote_ip`.

Reply:
I made a modification. We know that the real ip is always the first one in the `http_x_forwarded_for` variable so we check if the string starts with the ip we want. WDYT ?